pkg/network/setup/netstat_test.go

```diff
@@ -235,6 +235,54 @@ var _ = Describe("netstat", func() {
 		}), "the pod IP/s should be reported in the status")
 	})
 
+	It("should report SR-IOV interface when guest-agent is inactive and no other interface exists", func() {
+		const (
+			networkName = "sriov-network"
+			ifaceMAC    = "C0:01:BE:E7:15:G0:0D"
+		)
+
+		sriovIface := newVMISpecIfaceWithSRIOVBinding(networkName)
+		// The MAC is specified intentionally to illustrate that it is not reported if the GA is not present.
+		sriovIface.MacAddress = ifaceMAC
+		setup.addSRIOVNetworkInterface(
+			sriovIface,
+			newVMISpecMultusNetwork(networkName),
+		)
+
+		setup.NetStat.UpdateStatus(setup.Vmi, setup.Domain)
+
+		Expect(setup.Vmi.Status.Interfaces).To(Equal([]v1.VirtualMachineInstanceNetworkInterface{
+			newVMIStatusIface(networkName, nil, "", "", ""),
+		}), "the SR-IOV interface should be reported in the status, associated to the network")
+	})
+
+	It("should not report SR-IOV interface when guest-agent is inactive and a regular interface exists", func() {
```

Comment:
This sounds like a bug. The appearance of the sriov interface shouldn't be effected by another interface existence.

Reply:
Yes, SR-IOV reporting is buggy.
On one hand, it is reported sometimes even if there is no "proof" of it being there. On the other hand, it is not reported if at least one non-SRIOV interface is there.

None of these are correct, both represent a bug: When the guest has no guest-agent, SR-IOV interfaces are not reported correctly (either not at all, or based on false data).

But al least now we express this clearly through the tests.

I opened https://github.com/kubevirt/kubevirt/issues/7050 to track this.
Ref it also in the commit message.

Reply:
Can you please add the ref to #7050 to the code as well? So the next developer that will see those tests won't think it is "by design" behaviour.

Reply:
Sure, I'll add it to the tests.

Reply:
Done